### PR TITLE
Test timeout

### PR DESCRIPTION
**Pull request by arjunguha**

### CODE REVIEW

**Comment by arjunguha** [comment]
Some more testing is needed:

- [ ] A test where a higher order function has an infinite loop
- [ ] A test where a higher-order function runs for a while

**Comment by arjunguha** [comment]
@lchsam @joydeep-b as defined, the timeout now works in the browser. We should either have a longer timeout for students or disable the timeout by default. Which option?

**Comment by joydeep-b** [comment]
Until Project 1 is done, let's disable client-side.

File: ts/runtime.ts [hunk 1]
```diff
@@ -321,29 +321,47 @@ export function test(description: string, testFunction: () => void) {
...
+        runner.pause(() => {
+          if (done) { return; }
+          suspend.continuation = k;
+          suspend.onDone = onDone;
           tests.push({
+            failed: true,
+            error: 'time limit exceeded',
+            description: description
+          });
+          runner.resume();
```
[the remark this answers is not in the document]
Comment (arjunguha) [reply ts/runtime.ts]: Yes! This is the weird part. The runner has a continuation saved to resume execution (which would just continue the infinite loop). But, on lines 333 and 334, I set the saved continuation to the continuation of the test function, which is the next test case (if any).

**Pull request merged**